py-polars/polars/functions/range/date_range.py

```diff
@@ -389,11 +389,11 @@ def _warn_for_deprecated_date_range_use(
     if (
         isinstance(start, datetime)
         or isinstance(end, datetime)
-        or "s" in interval
-        or "h" in interval
-        or ("m" in interval and "mo" not in interval)
         or time_unit is not None
         or time_zone is not None
+        or ("h" in interval)
+        or ("m" in interval and "mo" not in interval)
```

Comment:
Could you also update this one? I sort-of accounted for "mo" but your logic is better.

```suggestion
        or ("m" in interval.replace("mo", ""))
```